tests/test_preprocessor.py

```diff
@@ -32,6 +32,20 @@
 
 class DatasetPreProcessorTest(unittest.TestCase):
 
+    def test_time_must_be_expanded_time_bounds_as_var(self):
+        ds = new_test_dataset(day=5,
+                              add_time_bnds=True,
+                              add_time_as_dim_to_vars=False,
+                              time_bounds_name='date_bnds',
+                              add_time_bounds_as_var=True)
+        pre_processor = DatasetPreProcessor(input_concat_dim='time')
+        ds = pre_processor.preprocess_dataset(ds)
+        self.assertIn('date_bnds', ds)
+        self.assertIn('date_bnds', ds.coords)
+        self.assertEqual(('time', 'bnds'), ds.date_bnds.dims)
+        self.assertIn('r_ui16', ds)
+        self.assertEqual(('time', 'lat', 'lon'), ds.r_ui16.dims)
```

Comment:
You should still assert that the "time" coordinate

1. is in ds, 
2. a 1-D array, 
3. has "bounds" attribute and its value is "date_bnds".